src/substrate-lib/components/SubstrateLoadingAndErrorHandlingWrapper/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { useDispatch } from 'react-redux';
+import { toast, ToastContainer } from 'react-toastify';
+
+import { isBridgeLoaded } from '@/common/actions/general.actions';
+import FullLoadingSpinner from '@/components/FullLoadingSpinner';
+import InterlayHelmet from '@/parts/InterlayHelmet';
+import { ActionType, ApiStatus, KeyringStatus, useSubstrateState } from '@/substrate-lib/substrate-context';
+
+interface SubstrateLoadingAndErrorHandlingWrapperProps {
+  children: React.ReactNode;
+}
+
+const SubstrateLoadingAndErrorHandlingWrapper = ({
+  children
+}: SubstrateLoadingAndErrorHandlingWrapperProps): JSX.Element => {
+  const dispatch = useDispatch();
+
+  const { apiStatus, apiError, keyringStatus, keyring, api } = useSubstrateState();
+
+  // TODO: translate `FullLoadingSpinner` texts once confirmed
+  switch (apiStatus) {
+    case ApiStatus.Idle:
+    case ApiStatus.ConnectInit:
+    case ApiStatus.Connecting:
+      return <FullLoadingSpinner text='Connecting to Bridge' />;
+    case ApiStatus.Ready:
+      // TODO: remove `isBridgeLoaded` & `bridgeLoaded` use cases via another PR
+      dispatch(isBridgeLoaded(true));
```

Comment:
Logically, this is not needed but added here for compatibility as this flag is used in other components. By this point, the bridge API is instantiated for sure and before that, the loading UI/UX is displayed.